src/main_thread/tracks_store/tracks_store.ts

```diff
@@ -1268,34 +1430,38 @@ export default class TracksStore extends EventEmitter<ITracksStoreEvents> {
         continue;
       }
 
-      const audioAdaptation = getSupportedAdaptations(period, "audio")[0];
-      trackStorePeriod.audio.storedSettings =
-        audioAdaptation === undefined
-          ? null
-          : {
-              adaptation: audioAdaptation,
-              switchingMode: this._defaultAudioTrackSwitchingMode,
-              lockedRepresentations: new SharedReference<IRepresentationsChoice | null>(
-                null,
-              ),
-            };
+      const audioAdaptation = getSupportedAdaptations(period, "audio")[0] as
+        | IAdaptationMetadata
+        | undefined;
```

Comment:
The `as` is just to indicate that the value may be `undefiined`?
There may be better ways no? The problem I see with `as` is possible bugs if we make code changes while trusting TypeScript a little too much to save our back

Reply:
I'm wondering if TypeScript automatically translate `at` calls